src/pages/api/organize/[orgId]/journeys/[journeyId]/index.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// TODO: re-enable eslint
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import { createApiFetch } from 'utils/apiFetch';
+import { getMessages } from 'utils/locale';
+import { getTagMetadata } from './getTagMetadata';
+
+import { dummyTableData } from 'components/journeys/JourneyInstancesDataTable/index.spec';
+
+const getJourneyTableData = async (
+  req: NextApiRequest & { query: Record<string, string> },
+  res: NextApiResponse
+): Promise<void> => {
+  const {
+    query: { orgId, journeyId },
+    method,
+  } = req;
+  const apiFetch = createApiFetch(req.headers);
+
+  // Return error if method other than GET
+  if (method !== 'GET') {
+    res.setHeader('Allow', ['GET']);
+    res.status(405).end(`Method ${method} Not Allowed`);
+    return;
+  }
+
+  try {
+    // const journeyInstancesRes = await apiFetch(
+    //   `/orgs/${orgId}/journeys/${journeyId}/instances`
+    // );
+    // const { data: journeyInstances } = await journeyInstancesRes.json();
+
+    // Retrieve column names
+    const columnNames = Object.fromEntries(
+      Object.entries(
+        await getMessages('en', ['pages.organizeJourneyInstances'])
+      ).map(([key, value]) => [key.split('.').pop(), value])
+    );
```

Comment:
I had to look at this for a while to understand what was happening here but now I understand! Does this have to happen on the backend? I would imagine the static columns being defined statically, one by one, on the frontend and components like `<FormattedMessage>` being used explicitly there. I'm not sure what the current solution means for things like missing messages/translations etc, but I know that would be handled gracefully with the regular components.

Reply:
It's because getMessages is async - hence prefetched on server. Could re-engineer but since it seems harmless I think let's leave it for now.